pylint/testutils/functional/find_functional_tests.py

```diff
@@ -86,7 +90,6 @@ def walk(path: Path) -> Iterator[Path]:
         )
         for _file_or_dir in parent_dir_files:
             if _file_or_dir.is_dir():
-                _files = _get_files_from_dir(_file_or_dir, violations)
```

Comment:
`_files` here isn't used anywhere. Instead this line lead to a duplicate warning for subpackages which are walked a few lines below anyway. Example output from one of the tests.
```py
The following directory contains too many functional tests files:
- /.../pylint/tests/testutils/data/m: 3 when the max is 1
- /.../pylint/tests/testutils/data/m/max_overflow: 2 when the max is 1
- /.../pylint/tests/testutils/data/m/max_overflow: 2 when the max is 1
```